presto-main/src/main/java/com/facebook/presto/metadata/FunctionRegistry.java

```diff
@@ -555,6 +562,10 @@ public static boolean canCoerce(Type actualType, Type expectedType)
         if (expectedType.equals(actualType)) {
             return true;
         }
+        // is this a coercion of the type only (no data change)
```

Comment:
why not export canCoerce, isTypeOnlyCoercion, getCommonSuperType to a separate class (e.g. TypeHierarchy). Even better would be to make type hierarchy expendable by providing plugin/delegate mechanism?